i-VRESSE/haddock3-webapp
# feat: save builder definitions in indexedDB so it can be reloaded upon return

# Save builder content

It saves workflow definitions to indexedDB in the "builder" objectStore when user navigates away from builder and it loads these back.

## What is implemented?
- created custom hook useIndexedDb to load saved builder definitions. The hook uses methods in indexedDB.ts file
- created custom hook useSaveOnLeave to save builder definitions when users leaves builder route
- increased indexedDB version to 2 in order to create additional objectStore "builder". The database have now 2 object stores "zips" and "builder".

## How to test?
- Without logging in navigate to builder. Create some definitions and navigate away to some other app page. Currently there will be console log saying "useSaveOnLeave...OK" if definitions were saved succefully.
- Navigate back to builder and confirm that definitions you had are properly loaded from indexedDB.
- Login and test the process again.
- Try loading scenarios. Note! I was not able to test this approach because I do not know how to trigger scenario "process".  But the code should do the following: if there is scenario defined to load in builder (zips objectStore) the scenario will be loaded. If there is no scenario to load we look if there are builder definitions
